docs/en/middleware/zenoh.md

```diff
@@ -0,0 +1,194 @@
+
+The following guide describes the architecture and various options for setting up the Zenoh client and router.
+In particular, it covers the options that are most important to PX4 users exploring Zenoh as an alternative communication layer for ROS 2.
+
+## Architecture
+
+The Zenoh-based middleware consists of a client running on PX4 and a Zenoh router running on the companion computer, with bi-directional data exchange between them over a UART, TCP, UDP, or multicast-UDP link.
+The router acts as a broker and discovery service, enabling PX4 to publish and subscribe to topics in the global Zenoh data space.
+This allows seamless integration with ROS 2 nodes using [`rmw_zenoh`](https://github.com/ros2/rmw_zenoh), and supports flexible deployment across distributed systems.
+
+![Architecture PX4 Zenoh-Pico with ROS 2](../../assets/middleware/zenoh/architecture-px4-zenoh.svg)
+
+The client is the _PX4 Zenoh-Pico Node_ referred to above, which is implemented in the [PX4 `zenoh` module](../modules/modules_driver.md#zenoh).
+This is based on Zenoh-Pico, a minimalistic version of [Eclipse Zenoh](https://zenoh.io/) (a data-centric protocol designed for real-time, distributed, and resource-constrained environments).
+
+The router suggested above is [zenohd](https://github.com/eclipse-zenoh/zenoh/tree/main/zenohd).
+
+:::info
+UART is supported by Zenoh but has not yet implemented in the PX4 Zenoh-Pico node.
+:::
+
+## ROS 2 Zenoh Bring-up on Linux Companion
+
+In order for PX4 uORB topics to be shared with ROS 2 applications, you will need the PX4 Zenoh-Pico Node client running on your FMU, connected to a Zenoh router running on the companion computer (or elsewhere in the network).
+
+First select Zenoh as the ROS 2 transport by setting the `RMW_IMPLEMENTATION` environment variable as shown:
+
+```sh
+export RMW_IMPLEMENTATION=rmw_zenoh_cpp
+```
+
+Then start the Zenoh router using the command:
+
+```sh
+ros2 run rmw_zenoh_cpp rmw_zenohd
+```
+
+For more information about the Zenoh Router see the [rmw_zenoh](https://github.com/ros2/rmw_zenoh?tab=readme-ov-file#start-the-zenoh-router) documentation.
+
+## PX4 Zenoh-Pico Node Setup
+
+### PX4 Firmware
+
+Before setting up the Zenoh communication, first make sure that your firmware contains the driver that implements the [`zenoh` driver](../modules/modules_driver.html#zenoh), which provides the implementation of the _PX4 Zenoh-Pico Node_.
+
+You can check if the module is present on your board by searching for the key `CONFIG_MODULES_ZENOH=y` in your board's `default.px4board` KConfig file.
+For example, you can see that the module is present in `px4_fmu-v6xrt` build targets from [/boards/px4/fmu-v6xrt/default.px4board](https://github.com/PX4/PX4-Autopilot/blob/main/boards/px4/fmu-v6xrt/default.px4board#L91).
+If it is not present you can add this key to your board configuration and rebuild.
```

Comment:
Section rewritten so people can try this on any board if that is what they have.

If I did this for fmu-v5 build target, would I need to:
- Remove the uXRCE-DDS module?
- Any thing else to get it working

Essentially, what level of coexistence can this have with the DDS alternative?

Reply:
> You can check if the module is present on your board by searching for the key `CONFIG_MODULES_ZENOH=y` in your board's `default.px4board` KConfig file.

Indeed the text above described it correctly. Overall both Zenoh and uXRCE-DDS are perfectly capable to co-exist. However simply due to flash constraints they can't be integrated in the same build. The targets in the table below have enough flash and thus have it enabled in their default builds.

Reply:
That is worth adding. Will add tomorrow. Thanks!

Reply:
Done in https://github.com/PX4/PX4-Autopilot/pull/25737/commits/b8790248c1a2cd50aecd7b0c61aa6d2a06363c5d